resource/deploymentService.js

```diff
@@ -0,0 +1,126 @@
+        await buildUtil.makeBuildDir(tokenId);
+        await buildUtil.prepContractDir(tokenId);
+        await buildUtil.buildCrowdsaleContract(tokenId);
+
+        let tokenObject = await tokenService.getTokenByTokenId(tokenId);
+        let tokenJSON = tokenObject.toJSON();
+
+        let crowdsaleObject = await crowdsaleService.getCrowdsaleByTokenId(tokenId);
+        let crowdsaleJSON = crowdsaleObject.toJSON();
+
+        await buildUtil.buildTokenContract(tokenId, tokenJSON.name, tokenJSON.symbol);
+
+        await buildUtil.prepMergeDir(tokenId);
+        await buildUtil.mergeCrowdsaleContract(tokenId);
+        let res = await buildUtil.deployCrowdsaleContract(tokenId, crowdsaleJSON);
+        LiveCycleUtil.finished(deployment, res);
+        await deployment.save();
+        await clientCallback(deployment);
+    } catch (e) {
+        log.error('contract deployment failed', e);
+        LiveCycleUtil.failed(deployment, e.message);
+        await deployment.save();
+    }
+}
+
+async function clientCallback(deployment) {
+    let url = deployment.get('callbackUrl');
+    let authToken = deployment.get('callbackAuthToken');
+    if (!url) {
+        return;
+    }
+    let jsn = deployment.toJSON();
+    let option = {
+        url: url,
+        json: {
+            deploymentId: jsn.objectId,
+            status: jsn.status,
+            transactionHash: jsn.transactionHash
+        }
+    };
+    if (jsn.status == 'failed') {
+        option.json.failureReason = jsn.failureReason;
+    }
+    if (authToken) {
+        option.headers = {
+            Authorization: authToken
+        }
+    }
+    request.post(option, function optionalCallback(err) {
+        if (err) {
+            return log.error('client callback failed', err);
+        }
+        log.log('client callback successfull');
+    });
+
+}
+
+exports.getDeploymentById = getById = async function(deploymendId) {
+    const deploymentQuery = new Parse.Query(ContractDeployment);
+    deploymentQuery.equalTo('objectId', deploymendId);
+    return await deploymentQuery.first();
+}
+
+const LiveCycleUtil = {
+        registered: function registered(deployment) {
+            deployment.set('status', 'registered');
+            deployment.set('createDate', new Date());
+        },
+        started: function (deployment) {
+            deployment.set('status', 'started');
+            deployment.set('updateDate', new Date());
+        },
+        failed: function (deployment, reason) {
+            deployment.set('status', 'failed');
+            deployment.set('failureReason', reason);
+            deployment.set('updateDate', new Date());
+        },
+        finished: function (deployment, deployResult) {
+            deployment.set('status', 'finished');
+            deployment.set('updateDate', new Date());
+            deployment.set('transactionHash', deployResult.transactionHash)
+            deployment.set('tokenAddress', deployment.address);
```

Comment:
when a crowdsale contract is deployed: actually there will two contract addresses will be returned:

1. crowdSale contract address;
2. token contract address.

we probably need to save both here. especially the crowdSale contract address. Because you can find token contract address, with the crowdSale contract address.